pytensor/configdefaults.py

```diff
@@ -1221,27 +1149,27 @@ def add_numba_configvars():
     )
 
 
-def _default_compiledirname():
+def _default_compiledirname() -> str:
     formatted = config.compiledir_format % _compiledir_format_dict
     safe = re.sub(r"[\(\)\s,]+", "_", formatted)
     return safe
 
 
-def _filter_base_compiledir(path):
+def _filter_base_compiledir(path: Path) -> Path:
     # Expand '~' in path
-    return os.path.expanduser(str(path))
+    return path.expanduser()
 
 
-def _filter_compiledir(path):
+def _filter_compiledir(path: Path) -> Path:
     # Expand '~' in path
-    path = os.path.expanduser(path)
+    path = path.expanduser()
     # Turn path into the 'real' path. This ensures that:
     #   1. There is no relative path, which would fail e.g. when trying to
     #      import modules from the compile dir.
     #   2. The path is stable w.r.t. e.g. symlinks (which makes it easier
     #      to re-use compiled modules).
-    path = os.path.realpath(path)
-    if os.access(path, os.F_OK):  # Do it exist?
+    path = path.resolve()
+    if path.exists():
         if not os.access(path, os.R_OK | os.W_OK | os.X_OK):
```

Comment:
This could be changed as well to use `path.stat().st_mode`, to avoid using `os.access`, but it might not be worth it.

Reply:
Yes indeed, but not sure it's worth it indeed. I think it would make this code more verbose.